examples/targeting/add_customer_negative_criteria.rb

```diff
@@ -0,0 +1,111 @@
+#           You may obtain a copy of the License at
+#
+#           http://www.apache.org/licenses/LICENSE-2.0
+#
+#           Unless required by applicable law or agreed to in writing, software
+#           distributed under the License is distributed on an "AS IS" BASIS,
+#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+#           implied.
+#           See the License for the specific language governing permissions and
+#           limitations under the License.
+#
+# This example adds various types of negative criteria to a customer. These
+# criteria will be applied to all campaigns for the customer.
+#
+#
+require 'optparse'
+require 'google/ads/google_ads'
+
+def add_customer_negative_criteria(customer_id)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  tradgedy_criterion = client.resource(:CustomerNegativeCriterion)
+  tradgedy_criterion.content_label = client.resource(:ContentLabelInfo)
+  tradgedy_criterion.content_label.type = :TRAGEDY
+
+  placement_criterion = client.resource(:CustomerNegativeCriterion)
+  placement_criterion.placement = client.resource(:PlacementInfo)
+  placement_criterion.placement.url = client.wrapper.string("http://example.com")
+
+  ops = [
+    {create: tradgedy_criterion},
+    {create: placement_criterion},
+  ]
+
+  customer_negative_criterion_service = client.service(
+    :CustomerNegativeCriterion
+  )
+
+  response = customer_negative_criterion_service.mutate_customer_negative_criteria(customer_id, ops)
+
+  if response.partial_failure_error != nil
+    raise "Got a partial failure #{response.partial_failure_error.inspect} while creating criterea"
+  end
+
+  # Display the results.
+  response.results.each do |negative_criterion|
+    puts "Customer negative criterion with criterion resource name #{negative_criterion.resource_name} was added"
+  end
+end
+
+if __FILE__ == $0
+  options = {}
+  # The following parameter(s) should be provided to run the example. You can
+  # either specify these by changing the INSERT_XXX_ID_HERE values below, or on
+  # the command line.
+  #
+  # Parameters passed on the command line will override any parameters set in
+  # code.
+  #
+  # Running the example with -h will print the command line usage.
+  options[:customer_id] = 'INSERT_CUSTOMER_ID_HERE'
+
+  OptionParser.new do |opts|
+    opts.banner = sprintf('Usage: add_campaigns.rb [options]')
+
+    opts.separator ''
+    opts.separator 'Options:'
+
+    opts.on('-C', '--customer-id CUSTOMER-ID', String, 'Customer ID') do |v|
+      options[:customer_id] = v
+    end
+
+    opts.separator ''
+    opts.separator 'Help:'
+
+    opts.on_tail('-h', '--help', 'Show this message') do
+      puts opts
+      exit
+    end
+  end.parse!
+
+  begin
+    add_customer_negative_criteria(options.fetch(:customer_id))
```

Comment:
The other examples use options[:customer_id]; is there a reason you used fetch instead?

Reply:
`fetch` is generally preferable because it raises if a value is not present: https://ruby-doc.org/core-2.2.0/Hash.html#method-i-fetch. It's generally considered best practice to use `fetch` in all places.

Reply:
Can you add this (and any other style/best-practices things you notice) to that doc you shared with me about potential improvements? Then feel free to resolve this.

Reply:
Done!